pkg/cfn/builder/managed_launch_template_test.go

```diff
@@ -44,8 +45,9 @@ var _ = Describe("ManagedNodeGroup builder", func() {
 		fakeVPCImporter.SecurityGroupsReturns(gfnt.Slice{gfnt.MakeFnImportValueString("eksctl-lt::ClusterSecurityGroupId")})
 		fakeVPCImporter.SubnetsPublicReturns(gfnt.MakeFnSplit(",", gfnt.MakeFnImportValueString("eksctl-lt::SubnetsPublic")))
 
-		stack := NewManagedNodeGroup(provider.MockEC2(), clusterConfig, m.ng, NewLaunchTemplateFetcher(provider.MockEC2()), false, fakeVPCImporter)
-		stack.UserDataMimeBoundary = "//"
+		bootstrapper := nodebootstrap.NewManagedAL2Bootstrapper(m.ng)
```

Comment:
I think this should be a fake bootstrapper here

Reply:
A fake bootstrapper would make the test less useful. This is more of an E2E unit test that asserts that for a given nodegroup, the correct CloudFormation template (which includes user data) is generated.

Reply:
we should be able to test that in a the nodebootstrap unit, having all these things coupled unnecessarily is causing problems when we try to refactor/add new things

Reply:
This test is for testing the launch template generation, an important part of which is the user data, for a given nodegroup. It predates the bootstrapper interface. I would have liked to have a test that matches the real world, but seeing as I have added tests for the managed bootstrapper unit separately, I think I agree that this can be replaced with a fake.